### PR TITLE
Allowing duplicates in the FileDataSource to be ignored

### PR DESCRIPTION
**Pull request by JeffAshton**
**Requirements**

- [X] I have added test coverage for new or changed functionality
- [X] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [X] I have validated my changes against all supported platform versions

**Describe the solution you've provided**

When rolling files in a safe manner, `FileDataSource` always throws exceptions.

**Describe alternatives you've considered**

See https://github.com/launchdarkly/dotnet-server-sdk/pull/120

### CODE REVIEW

File: src/LaunchDarkly.ServerSdk/Files/FlagFileData.cs [hunk 1]
```diff
@@ -17,46 +18,58 @@ class FlagFileData
         [JsonProperty(PropertyName = "segments", NullValueHandling = NullValueHandling.Ignore)]
         public Dictionary<string, JToken> Segments { get; set; }
 
-        public void AddToData(IDictionary<IVersionedDataKind, IDictionary<string, IVersionedData>> allData)
+        public void AddToData(IDictionary<IVersionedDataKind, IDictionary<string, IVersionedData>> allData,
+            DuplicateKeysHandling duplicateKeysHandling)
```
Comment (eli-darkly) [inline src/LaunchDarkly.ServerSdk/Files/FlagFileData.cs]: How about making `duplicateKeysHandling` a field that's set in the constructor, so you don't have to keep passing it to `AddItem`?

Comment (JeffAshton) [reply src/LaunchDarkly.ServerSdk/Files/FlagFileData.cs]: The relationship between `FlagFileData` and `FlagFileParser` makes this awkward.  I pulled out the `AddToData` method into its own class in a alternate PR:

https://github.com/launchdarkly/dotnet-server-sdk/pull/123/files

Comment (eli-darkly) [reply src/LaunchDarkly.ServerSdk/Files/FlagFileData.cs]: Oh yeah, I see what you mean. Sorry, I wrote this code but it's been a while.

At this point though, I'm finding it slightly hard to follow three nearly identical PRs for the same feature (especially since Github does not make it easy to see at a glance what the differences are between them). Could you please not open any more PRs for this in the near future, and just pick one to focus on— if you have any more proposed changes, just make them on the same branch and then if they're not desirable they can easily be reverted. I think the newest one, 123, is what I would prefer.
